core/http-auth-event-stream/src/main/java/software/amazon/awssdk/http/auth/eventstream/internal/DefaultAwsV4EventStreamHttpSigner.java

```diff
@@ -27,15 +40,119 @@
+        return CompletableFuture.completedFuture(HTTP_CONTENT_SHA_256);
+    }
+
+    @Override
+    public void addPrerequisites(SdkHttpRequest.Builder requestBuilder,
+                                 ContentChecksum contentChecksum, AwsV4HttpProperties properties) {
+        requestBuilder.putHeader(X_AMZ_CONTENT_SHA256, HTTP_CONTENT_SHA_256);
+        v4Signer.addPrerequisites(requestBuilder, contentChecksum, properties);
+    }
+
+    @Override
+    public CanonicalRequestV2 createCanonicalRequest(SdkHttpRequest request, ContentChecksum contentChecksum,
+                                                     AwsV4HttpProperties properties) {
+        return v4Signer.createCanonicalRequest(request, contentChecksum, properties);
+    }
+
+    @Override
+    public String createSignString(String canonicalRequestHash, AwsV4HttpProperties properties) {
+        return v4Signer.createSignString(canonicalRequestHash, properties);
+    }
+
+    @Override
+    public byte[] createSigningKey(AwsV4HttpProperties properties) {
+        return v4Signer.createSigningKey(properties);
+    }
+
+    @Override
+    public String createSignature(String stringToSign, byte[] signingKey, AwsV4HttpProperties properties) {
+        return v4Signer.createSignature(stringToSign, signingKey, properties);
+    }
+
+    @Override
+    public void addSignature(SdkHttpRequest.Builder requestBuilder,
+                             CanonicalRequestV2 canonicalRequest,
+                             String signature,
+                             AwsV4HttpProperties properties) {
+        v4Signer.addSignature(requestBuilder, canonicalRequest, signature, properties);
+    }
+
+    @Override
+    public ContentStreamProvider processPayload(ContentStreamProvider payload,
+                                                SigV4RequestContext v4RequestContext, AwsV4HttpProperties properties) {
+        return v4Signer.processPayload(payload, v4RequestContext, properties);
+    }
+
+    @Override
+    public Publisher<ByteBuffer> processPayload(Publisher<ByteBuffer> payload,
+                                                CompletableFuture<SigV4RequestContext> futureV4RequestContext,
+                                                AwsV4HttpProperties properties) {
+        if (payload == null) {
+            return null;
+        }
+
+        // The request future will always be completed when we access it here, because it doesn't depend on the payload since
+        // we override the `createContentHash` method with an already completed future
+        return futureV4RequestContext.thenApply(
+            v4RequestContext -> new SigV4DataFramePublisher(
+                payload,
+                properties.getCredentials(),
+                properties.getCredentialScope(),
+                v4RequestContext.getSignature(),
+                properties.getSigningClock()
+            )
+        ).join();
```

Comment:
No, this stays. The publisher needs the signature, which it can only get from signing the request. Otherwise, this method would need to return a future publisher, which doesn't seem right, because a publisher is async by nature. The reasoning is stated in the comment above. 

It's sort of a chicken-and-egg situation if we try and support flexible checksum here, because the payload signing needs the request signature, which means (in the case of flexible checksum) the entire payload must already be processed in order to calculate the hash to use in the signature. Flexible checksum isn't supported in the old implementation of event-stream-signer, so kept it consistent here. Otherwise, I suppose we could have the publisher accept a signature-future instead?